pkg/ebpf/events_pipeline.go

```diff
@@ -428,3 +432,21 @@ func (t *Tracee) handleError(err error) {
 	t.stats.ErrorCount.Increment()
 	logger.Error("tracee encountered an error", "error", err)
 }
+
+// parseArguments is required by the rules engine,
+// if the rule engine staged is enabled, parseArguments will happen there,
+// before sending an event to be processed, if the rule engine is disabled,
+// parseArgumnets happens on the sink stage (because anyone pipeing the output to tracee-rules will need it)
```

Comment:
I tried to rewrite this but got confused. Could you elaborate on separate sentences?